homeassistant/components/history_stats/coordinator.py

```diff
@@ -0,0 +1,96 @@
+"""History stats data coordinator."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from homeassistant.core import CALLBACK_TYPE, Event, HomeAssistant, callback
+from homeassistant.exceptions import TemplateError
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.start import async_at_start
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .data import HistoryStats, HistoryStatsState
+
+_LOGGER = logging.getLogger(__name__)
+
+
+UPDATE_INTERVAL = timedelta(minutes=1)
+
+
+class HistoryStatsUpdateCoordinator(DataUpdateCoordinator):
+    """DataUpdateCoordinator to gather data for a specific TPLink device."""
+
+    data: HistoryStatsState
```

Comment:
The DataUpdateCoordinator is generic so we can type `data` in the class definition above.

Reply:
https://github.com/home-assistant/core/pull/71479